src/containers/AssetViewer.tsx

```diff
@@ -82,112 +83,96 @@ export class AssetViewer extends React.Component<Props, State> {
-    return Number(
-      Object.keys(representsAsset).find(
-        assetId =>
-          representsAsset[Number(assetId)].modelId === modelId &&
-          representsAsset[Number(assetId)].revisionId === revisionId
-      )
-    );
+    return undefined;
   }
 
   getNodeId = (fetchIfMissing: boolean) => {
-    const { assetMappings } = this.props;
-    if (assetMappings.byAssetId[this.assetId]) {
-      const mapping = assetMappings.byAssetId[this.assetId];
-      return mapping.nodeId;
+    const {
+      app: { modelId, revisionId, nodeId },
+    } = this.props;
+    if (nodeId) {
+      return nodeId;
     }
 
-    if (fetchIfMissing && this.props.modelId && this.props.revisionId) {
-      const { modelId, revisionId } = this.props;
+    if (this.assetId && fetchIfMissing && modelId && revisionId) {
       this.props.doFetchMappingsFromAssetId(modelId, revisionId, this.assetId);
     }
 
-    return null;
-  };
-
-  getAsset = () => {
-    const { assets } = this.props;
-
-    return assets.all[this.assetId];
+    return undefined;
   };
 
   render3D = () => {
     const { rootAssetId, assetId } = this;
-    const { nodeId: propNodeId } = this.props;
+    const { nodeId: propNodeId, modelId, revisionId } = this.props.app;
     const nodeId = propNodeId || this.getNodeId(false);
     return (
       <Model3D
-        assetId={assetId}
-        modelId={this.props.modelId!}
-        revisionId={this.props.revisionId!}
+        assetId={assetId!}
+        modelId={modelId!}
+        revisionId={revisionId!}
         nodeId={nodeId}
         onAssetIdChange={(id: number) =>
-          this.props.onAssetIdChange(rootAssetId, id)
+          this.props.setAssetId(rootAssetId!, id)
+        }
+        onNodeIdChange={(id: number) =>
+          this.props.setModelAndRevisionAndNode(modelId!, revisionId!, id)
         }
-        onNodeIdChange={(id: number) => this.props.onNodeIdChange(id)}
         cache={this.cache}
       />
     );
   };
 
   renderPNID = () => {
-    const { rootAssetId } = this;
-    const asset = this.getAsset();
-    return (
-      <PNIDViewer
-        asset={asset}
-        onAssetIdChange={(id: number) =>
-          this.props.onAssetIdChange(rootAssetId, id)
-        }
-      />
-    );
+    const { asset } = this;
+    if (!asset) {
+      return null;
+    }
+    return <PNIDViewer />;
   };
 
   renderAssetNetwork = () => {
     const { rootAssetId } = this;
+    if (!rootAssetId) {
+      return null;
+    }
     return (
       <div className="bottom">
         <AssetNetworkViewer
-          rootAssetId={rootAssetId}
-          asset={this.getAsset()}
           topShowing={this.props.show3D || this.props.showPNID}
-          onAssetIdChange={(id: number) =>
-            this.props.onAssetIdChange(rootAssetId, id)
-          }
         />
       </div>
     );
   };
 
   renderRelationshipsViewer = () => {
     const { rootAssetId } = this;
+    if (!rootAssetId) {
```

Comment:
Should we only show it if user has access to relationships?

Reply:
Good point, we should add that in along with other permissions, we can track it here https://cognitedata.atlassian.net/browse/CONTEXT-257

Reply:
tracking here
https://cognitedata.atlassian.net/browse/CONTEXT-257

we should also check on other permissions